### DISCUSSION TITLE
I can't build Directus : FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory

### DISCUSSION THREAD

**Discussion by Powma** [directus/directus#14348] [Archive]
### Describe the Bug

Hello, 

I have 16go of memory in my laptop and I think Directus have a dependance who do a leak memory or maybe the problem is NodeJS.

### To Reproduce

git clone git@github.com:directus/directus.git
cd directus
npm install
npm run build

### Errors Shown

<--- Last few GCs --->

[8716:000001C8678F2A90]    54515 ms: Scavenge 2022.2 (2076.0) -> 2017.4 (2078.0) MB, 6.3 / 0.0 ms  (average mu = 0.739, current mu = 0.411) allocation failure
[8716:000001C8678F2A90]    54541 ms: Scavenge 2024.0 (2078.0) -> 2019.0 (2079.5) MB, 6.9 / 0.0 ms  (average mu = 0.739, current mu = 0.411) allocation failure
[8716:000001C8678F2A90]    54693 ms: Scavenge 2025.5 (2079.5) -> 2020.5 (2096.5) MB, 134.2 / 0.0 ms  (average mu = 0.739, current mu = 0.411) allocation failure

<--- JS stacktrace --->

FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
 1: 00007FF69B1E7C4F v8::internal::CodeObjectRegistry::~CodeObjectRegistry+114207
 2: 00007FF69B175EC6 DSA_meth_get_flags+65542
 3: 00007FF69B176D7D node::OnFatalError+301
 4: 00007FF69BAAB6CE v8::Isolate::ReportExternalAllocationLimitReached+94
 5: 00007FF69BA95CAD v8::SharedArrayBuffer::Externalize+781
 6: 00007FF69B93907C v8::internal::Heap::EphemeronKeyWriteBarrierFromCode+1468
 7: 00007FF69B936194 v8::internal::Heap::CollectGarbage+4244
 8: 00007FF69B933B10 v8::internal::Heap::AllocateExternalBackingStore+2000
 9: 00007FF69B958696 v8::internal::Factory::NewFillerObject+214
10: 00007FF69B68AB65 v8::internal::DateCache::Weekday+1797
11: 00007FF69BB393C1 v8::internal::SetupIsolateDelegate::SetupHeap+494417
12: 000001C86A432272

### What version of Directus are you using?

9.14.1

### What version of Node.js are you using?

16.16.0

### What database are you using?

nothing

### What browser are you using?

nothing

### How are you deploying Directus?

locally

**Comment by rijkvanzanten** [comment]
Node has a memory limit by default that doesn't match the memory available on the system by default. Even though you have 16GB of RAM, Node will still default to a lower memory limit (512MB if I'm not mistaken). Try setting `--max-old-space-size` to something like `8192` 👍🏻

**Reply by rijkvanzanten** [reply]
(We actually do the same in the CI: https://github.com/directus/directus/blob/4775cf33d92f8673afc2bd9feea18de28c767af4/.github/workflows/release.yml#L11)

**Comment by Powma** [comment]
Hello @rijkvanzanten 

Thanks for your answer, but maybe it's better to edit the file configuration `package.json` or explain in the documentation ?

I think the problem is not obvious, it's just my opinion.

**Comment by rijkvanzanten** [comment]
I don't think it's something we should configure for everybody through package.json, as it's value is based on whatever system you're running on. Speaking of system, what OS are you running the build under? I haven't ran into this issue outside of CI before

**Reply by Powma** [reply]
I use Windows 11, I compile with my terminal Windows, but maybe you have a better solution.

**Reply by rijkvanzanten** [reply]
`--max_old_space_size=8192` is the way to go 👍🏻 Node has a pretty low memory limit by default, and `vite` simply needs more when building an app of this size

**Reply by anlexN** [reply]
` NODE_OPTIONS=--max_old_space_size=8192 pnpm -r build`
I am success.
![image](https://user-images.githubusercontent.com/8686739/188308189-90968566-b06d-4d30-8298-13968d664685.png)

**But I think directus should use React, not vue.**

**Reply by anlexN** [reply]
Inside my country, the best-known companies: Alibaba, Baidu, Tencent **use React**, outside my country, the best-known companies: Microsoft, Google, Facebook, Amazon **use React**
### Big companies set standards
